### PR TITLE
authenticate: add device-enrolled page

### LINKED ISSUE #2891: feat: display enrollment confirmation page following device registration
**Is your feature request related to a problem? Please describe.**

It's not clear if a user has successfully enrolled their device with Pomerium. 

**Describe the solution you'd like**

- We should have successful registration endpoint (a under `.pomerium`) on the authenticate service that notifies the user that they've successfully enrolled their device. 
- We should show the user the device enrollment id (maybe a short hash which they can compare to admin logs? for any debugging needs) on the thank you screen.
- We should show any errors during enrollment to the end user if they can occur. 

**Describe alternatives you've considered**

**Explain any additional use-cases**

In the enterprise version of Pomerium, this would serve as a reasonable default redirect url following TOFU registration. 

**Additional context**

- https://github.com/pomerium/pomerium-console/issues/2232

### PR DESCRIPTION
**Pull request by calebdoxsey**
## Summary
Add a new device-enrolled page that can be shown after a device is enrolled successfully.

## Related issues
Fixes #2891 

## Checklist
- [x] reference any related issues
- [ ] updated docs
- [ ] updated unit tests
- [ ] updated UPGRADING.md
- [x] add appropriate tag (`improvement` / `bug` / etc)
- [x] ready for review

### CODE REVIEW

**Review by wasaga — APPROVED** [review]

File: authenticate/handlers.go [hunk 1]
```diff
@@ -98,6 +99,7 @@ func (a *Authenticate) mountDashboard(r *mux.Router) {
 	sr.Path("/sign_in").Handler(a.requireValidSignature(a.SignIn))
 	sr.Path("/sign_out").Handler(a.requireValidSignature(a.SignOut))
 	sr.Path("/webauthn").Handler(webauthn.New(a.getWebauthnState))
+	sr.Path("/device-enrolled").Handler(handlers.DeviceEnrolled())
```
Comment (desimone) [inline authenticate/handlers.go:102]: Should we ensure that the origin of the request is Pomerium (e.g. is valid signed / hmac'd?)

File: authenticate/handlers/webauthn/webauthn.go [hunk 2]
```diff
@@ -302,7 +302,17 @@ func (h *Handler) handleRegister(w http.ResponseWriter, r *http.Request, state *
...
+	// add the device credential ID to the redirect uri
+	redirectURI, err := urlutil.ParseAndValidateURL(redirectURIParam)
+	if err != nil {
+		return err
+	}
+	q := redirectURI.Query()
+	q.Set(urlutil.QueryDeviceCredentialID, deviceCredentialID)
+	redirectURI.RawQuery = q.Encode()
+
+	return h.saveSessionAndRedirect(w, r, state, redirectURI.String())
```
Comment (desimone) [inline authenticate/handlers/webauthn/webauthn.go:315]: I believe this creates a signed url right?

File: authenticate/handlers.go (hunk 1 above)
Comment (desimone) [reply authenticate/handlers.go:102]: It's becoming a little hard to follow all the different handlers getting mounted here. 

We have handlers in this directory, `webauthn` and now /handlers. Thoughts on adding these handler to the existing `webauthn` package?

File: authenticate/handlers/device-enrolled.go [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+	"net/http"
+
+	"github.com/pomerium/pomerium/internal/frontend"
+	"github.com/pomerium/pomerium/internal/httputil"
+	"github.com/pomerium/pomerium/internal/urlutil"
+)
+
+// DeviceEnrolled displays an HTML page informing the user that they've successfully enrolled a device.
+func DeviceEnrolled() http.Handler {
+	tpl := template.Must(frontend.NewTemplates())
```
Comment (desimone) [inline authenticate/handlers/device-enrolled.go:14]: Does this get evaluated on each call?

File: authenticate/handlers/device-enrolled.go [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+	type TemplateData struct {
+		DeviceCredentialID string
+	}
+	return httputil.HandlerFunc(func(w http.ResponseWriter, r *http.Request) error {
+		return tpl.ExecuteTemplate(w, "device-enrolled.html", TemplateData{
+			DeviceCredentialID: r.FormValue(urlutil.QueryDeviceCredentialID),
```
Comment (desimone) [inline authenticate/handlers/device-enrolled.go:20]: I might have missed it but it doesn't look like this url's HMAC is checked. It's probably not a huge deal but it could mean this value could be spoofed, and this url could be redirected to from another source besides pomerium.

**Review by desimone — CHANGES_REQUESTED** [review]

File: authenticate/handlers/webauthn/webauthn.go (hunk 2 above)
Comment (calebdoxsey) [reply authenticate/handlers/webauthn/webauthn.go:315]: I did not change the behavior of this.

File: authenticate/handlers/device-enrolled.go (hunk 3 above)
Comment (calebdoxsey) [reply authenticate/handlers/device-enrolled.go:14]: no

File: authenticate/handlers/device-enrolled.go (hunk 4 above)
Comment (calebdoxsey) [reply authenticate/handlers/device-enrolled.go:20]: It is not. If I HMAC it we will have a double-HMAC'd URL that will expire almost immediately. I can do this but it won't be useful for what this page was supposed to be since we're generating links which last a very long time. (the redirect URI is a component of the generated link)

File: authenticate/handlers.go (hunk 1 above)
Comment (calebdoxsey) [reply authenticate/handlers.go:102]: I don't think it belongs there, but sure I can move it.

File: authenticate/handlers/device-enrolled.go (hunk 4 above)
Comment (desimone) [reply authenticate/handlers/device-enrolled.go:20]: Understood. I was thinking that the webauthn handler would create a signed url on redirect. I get why that doesn't make sense if the endpoint is not Pomerium (and therefore doesn't want or need the hmac'd overhead). 

OK. Can an attacker craft a url that results in this page displaying an arbitrary crafted device credential id? Do we care? I guess we can accomplish the same thing by providing the user a link back to their user info page (`.pomerium`).

File: authenticate/handlers.go (hunk 1 above)
Comment (desimone) [reply authenticate/handlers.go:102]: It's okay.

**Review by desimone — CHANGES_REQUESTED** [review]
let's just KISS and get rid of displaying the `QueryDeviceCredentialID` in the go HTML template.
